traffic_portal/app/src/common/filters/EncodeURIComponentFilter.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This is a factory function for an AngularJS filter meant to allow easy
+ * sanitization of interpolated input into URL query strings.
+ *
+ * @example
+ *
+ * <a ng-href="https://example.test/something?foo={{someString | encodeURIComponent}}">
+ *   A safe link
+ * </a>
+ *
+ * @returns {(input: string) => string} A filter that sanitizes input text for safe insertion into URL query strings.
+ */
+
+var EncodeURIComponentFilter = function() {
```

Comment:
There shouldn't be a blank line between a JSDoc and the thing it documents.

I would feel worse about asking for such a small change if our CI was working, but fortunately (?) it isn't.

Also nit but I don't think you need to do `var Thing = function ...`, I'd be shocked if `function Thing ...` didn't work.

Reply:
Ok.. And other filters have `var thing = function`, so was keeping up with codebase. If we are changing in one, we should follow suit with other 3 filters too.

Reply:
That's fine. It doesn't matter, functionally.

Reply:
kewl.. all changes pushed.